pints/_log_likelihoods.py

```diff
@@ -117,6 +117,137 @@ def __call__(self, x):
                       - np.sum(autocorr_error**2, axis=0) / (2 * sigma**2))
 
 
+class GaussianIntegratedUniformLogLikelihood(pints.ProblemLogLikelihood):
+    r"""
+    Calculates a log-likelihood assuming independent Gaussian-distributed noise
+    at each time point where it is assumed that the dependence on
+    :math:`\sigma\sim U(a,b)` has been integrated out of the joint posterior,
```

Comment:
What's a joint posterior? Is this only applicable in a bayesian context then?

Reply:
I've added a bit more text at the bottom of the docstrings. The joint posterior is given below this text (in maths) and is described more in the notebook, so I personally think we are doing enough here.

Reply:
Also added more text at the top to help clarity (I hope!).